builder/dockerfile/internals.go

```diff
@@ -38,41 +37,42 @@ import (
 	"github.com/pkg/errors"
 )
 
-func (b *Builder) commit(id string, autoCmd strslice.StrSlice, comment string) error {
+func (b *Builder) commit(comment string) error {
 	if b.disableCommit {
 		return nil
 	}
 	if !b.hasFromImage() {
 		return errors.New("Please provide a source image with `from` prior to commit")
 	}
+	// TODO: why is this set here?
 	b.runConfig.Image = b.image
 
-	if id == "" {
-		cmd := b.runConfig.Cmd
-		b.runConfig.Cmd = strslice.StrSlice(append(getShell(b.runConfig), "#(nop) ", comment))
-		defer func(cmd strslice.StrSlice) { b.runConfig.Cmd = cmd }(cmd)
```

Comment:
There was an inconsistency here. The other two places that added a `#(nop)` comment did it was string concat, where as this adds it as another arg.

I'm not sure if this will cause a cache miss, but at least it's consistent now.